test/gui/workorder-details.lua

```diff
@@ -0,0 +1,155 @@
+-- test -dhack/scripts/devel/tests -tworkorder%-details
+
+config.mode = 'fortress'
+
+local gui = require('gui')
+local function send_keys(...)
+    local keys = {...}
+    for _,key in ipairs(keys) do
+        gui.simulateInput(dfhack.gui.getCurViewscreen(true), key)
+    end
+end
+
+local xtest = {} -- use to temporarily disable tests (change `function test.somename` to `function xtest.somename`)
+local wait = function(n)
+    --delay(n or 30) -- enable for debugging the tests
+end
+
+-- handle confirm plugin: we may need to additionally confirm order removal
+local confirm = require 'plugins.confirm'
+local confirmRemove = function() end
+if confirm.isEnabled() then
+    for _, c in pairs(confirm.get_conf_data()) do
+        if c.id == 'order-remove' then
+            if c.enabled then
+                confirmRemove = function()
+                    wait()
+                    -- without delays `confirm` can miss the key event
```

Comment:
are you sure? I have not found this to be the case in the past.

Reply:
it happens sporadically, when it does, removing order fails and test doesn't exit to map screen (and test fails at the `Test order should've been removed`-check). With `delay(5)` it happened once the first time I startet DF after PC restart -- I think it has something to do with anti-virus... Without the first `delay(5)` it happens every 3rd time or so.

Reply:
Let me look into this a little before merging. Otherwise code looks great!

Reply:
Thanks :)

The most important question: could you reproduce the issue?

Reply:
yes, reproduced, and I tracked down why. When the SELECT key is sent to the confirm prompt, the intercepted key isn't passed on to the underlying screen until the next rendered frame. Since we escape out of that screen before the next rendered frame, it is never sent -- until the screen is reopened. If you run the test, and it fails, and then you manually open the manager orders screen, you can see the test order there before it quickly disappears. If you pause the confirm dialog and resend MANAGER_REMOVE, it works. I'll put that suggestion in the PR.

Reply:
Is there a way to "wait until the next rendered frame"? It seems possible with `dfhack.timeout`, but probably cumbersome. On a related note, I can't find docs for the `delay` itself...

Reply:
Yes, but it's somewhat hacky. You can get how many ticks occur between each frame and wait that many ticks plus one. I'm not sure if this is always consistent, though, since the ticks per graphics frame is variable.

There are no docs for the unit tests, which is a bad thing. You can see the implementation of `delay` here: https://github.com/DFHack/dfhack/blob/0aa7ec877e4201af76b429438037fb5fc8a501c2/ci/test.lua#L82

It gets injected into the global environment for unit tests along with some other utility functions. In fact, you might be able to use `delay_until` to detect when a frame has been rendered.